### PR TITLE
onStep return value matters + side effect fixes

### LINKED ISSUE #2067: Creature::updateMapCache triggering onStepIn event
Because onStepIn was moved to Tile::queryAdd, Creature::updateMapCache is triggering it generating nasty side effects such as creature receiving damage when caching the map and there's a magic field in range.

How to reproduce:
1. Create a fire field (1487)
2. Summon a rat anywhere close to it

### PR DESCRIPTION
**Pull request by gugahoa**
Closes #2067 #2065

### CODE REVIEW

File: src/creature.cpp [hunk 1]
```diff
@@ -390,17 +390,13 @@ void Creature::onRemoveTileItem(const Tile* tile, const Position& pos, const Ite
 	}
 }
 
-void Creature::onCreatureAppear(Creature* creature, bool isLogin)
```
Comment (WibbenZ) [inline src/creature.cpp]: Is there even a need to keep the bool?

File: src/movement.cpp [hunk 2]
```diff
@@ -295,15 +295,15 @@ MoveEvent* MoveEvents::getEvent(const Tile* tile, MoveEvent_t eventType)
 	return nullptr;
 }
 
-uint32_t MoveEvents::onCreatureMove(Creature* creature, const Tile* tile, MoveEvent_t eventType)
+uint32_t MoveEvents::onCreatureMove(Creature* creature, const Tile* tile, const Position& fromPos, MoveEvent_t eventType)
```
Comment (WibbenZ) [inline src/movement.cpp]: Can't you use fromPosition?

File: src/luascript.h [hunk 3]
```diff
@@ -240,7 +240,7 @@ class LuaScriptInterface
 		bool pushFunction(int32_t functionId);
 
 		static int luaErrorHandler(lua_State* L);
-		bool callFunction(int params);
+		bool callFunction(int params, bool defaultReturn = false);
```
Comment (conde2) [inline src/luascript.h]: I Think that should be true for backwards compatibility.

Comment (gugahoa) [inline src/luascript.h]: It's true, but only when called from the onStep functions, for the rest it should be false

**Comment by gugahoa** [comment]
FLAG_PATHFINDING is the right one to use on ::queryAdd as it is what's passed when it's called only to cache, find path and so on, things that wouldn't trigger onStepIn even if ::queryAdd returned true because it would never call ::postAdd or ::postRemove)
But for the FLAG_NOLIMIT I completely agree with you that there should be a FLAG_IGNORESTEPOUT
If it's needed I can add it

**Comment by conde2** [comment]
What still left to do in this pull request?

**Comment by ranisalt** [comment]
Rebasing.

**Comment by conde2** [comment]
Is this guy active ? If not I can rebase it and references the credits to him.

**Comment by ranisalt** [comment]
Yes, he is. In fact he is a friend of mine and I'll notify him now.

**Comment by gugahoa** [comment]
Rebased. For now I can't test it against the bugs reported in the previous merge, if anyone could test it before me I would appreciate the help a lot. Thanks!

File: data/talkactions/scripts/reload.lua [hunk 4]
```diff
@@ -8,9 +8,6 @@ local reloadTypes = {
 	["channel"] = { targetType = RELOAD_TYPE_CHAT, name = "chatchannels" },
 	["chatchannels"] = { targetType = RELOAD_TYPE_CHAT, name = "chatchannels" },
 
-	["command"] = { targetType = RELOAD_TYPE_COMMANDS, name = "commands" },
-	["commands"] = { targetType = RELOAD_TYPE_COMMANDS, name = "commands" },
-
```
Comment (Mkalo) [inline data/talkactions/scripts/reload.lua]: Already fixed in #2172. Those commits looks pretty messy now maybe remake?

Comment (ranisalt) [inline data/talkactions/scripts/reload.lua]: It's because he merged instead of rebasing. It won't count as changed lines if merged, but it's ugly anyway.

**Pull request closed without merging**